requirements-dev.txt

```diff
@@ -6,6 +6,7 @@ cffi==1.14.5
 chardet==4.0.0
 click==8.0.1
 coverage==5.5
+craft-providers==0.0.1
```

Comment:
We mentioned this before and I don't remember what was decided back then. Craft-providers is GPLv3 and that should contaminate Charmcraft so it would need to be relicensed under GPLv3 as well -- or is there a different way to address that?

Reply:
For those curious:
```
$ pip-licenses 
 Name                 Version                    License                                             
 Jinja2               3.0.1                      BSD License                                         
 MarkupSafe           2.0.1                      BSD License                                         
 PyNaCl               1.4.0                      Apache License 2.0                                  
 PyYAML               5.4.1                      MIT License                                         
 appdirs              1.4.4                      MIT License                                         
 attrs                21.2.0                     MIT License                                         
 black                21.6b0                     MIT License                                         
 certifi              2021.5.30                  Mozilla Public License 2.0 (MPL 2.0)                
 cffi                 1.14.5                     MIT License                                         
 chardet              4.0.0                      GNU Library or Lesser General Public License (LGPL) 
 charmcraft           0.10.0+224.gb8b8ed2.dirty  Apache Software License                             
 click                8.0.1                      BSD License                                         
 coverage             5.5                        Apache Software License                             
 craft-providers      0.0.1                      GNU General Public License v3 (GPLv3)               
 flake8               3.9.2                      MIT License                                         
 humanize             3.9.0                      MIT License                                         
 idna                 2.10                       BSD License                                         
 iniconfig            1.1.1                      MIT License                                         
 jsonschema           3.2.0                      MIT License                                         
 macaroonbakery       1.3.1                      GNU Lesser General Public License v3 (LGPLv3)       
 mccabe               0.6.1                      MIT License                                         
 mypy-extensions      0.4.3                      MIT License                                         
 ops                  1.2.0                      Apache Software License                             
 packaging            20.9                       Apache Software License, BSD License                
 pathspec             0.8.1                      Mozilla Public License 2.0 (MPL 2.0)                
 pkg-resources        0.0.0                      UNKNOWN                                             
 pluggy               0.13.1                     MIT License                                         
 protobuf             3.17.3                     3-Clause BSD License                                
 py                   1.10.0                     MIT License                                         
 pyRFC3339            1.1                        MIT License                                         
 pycodestyle          2.7.0                      MIT License                                         
 pycparser            2.20                       BSD License                                         
 pydantic             1.8.2                      MIT License                                         
 pydocstyle           6.1.1                      MIT License                                         
 pyflakes             2.3.1                      MIT License                                         
 pymacaroons          0.13.0                     MIT License                                         
 pyparsing            2.4.7                      MIT License                                         
 pyrsistent           0.17.3                     MIT License                                         
 pytest               6.2.4                      MIT License                                         
 python-dateutil      2.8.1                      BSD License, Apache Software License                
 pytz                 2021.1                     MIT License                                         
 regex                2021.4.4                   Apache Software License                             
 requests             2.25.1                     Apache Software License                             
 requests-toolbelt    0.9.1                      Apache Software License                             
 requests-unixsocket  0.2.0                      Apache Software License                             
 responses            0.13.3                     Apache 2.0                                          
 six                  1.16.0                     MIT License                                         
 snowballstemmer      2.1.0                      BSD License                                         
 tabulate             0.8.9                      MIT License                                         
 toml                 0.10.2                     MIT License                                         
 typing-extensions    3.10.0.0                   Python Software Foundation License                  
 urllib3              1.26.5                     MIT License  
 ```

Looks like the two projects using GPL are actually LGPL.

Reply:
But @cmatsuoka 's point is still valid, right?

Reply:
LGPL is fine, but GPL looks problematic to me.

Reply:
I'll defer to @sergiusens, I don't think we had a final resolution on the previous discussion.  We can change the license for craft-providers if desired.

Reply:
I bumped the version to 0.0.2 with the corrected license.